library/general/src/lib/yast2/group.rb

```diff
@@ -0,0 +1,49 @@
+require "yast2/execute"
+
+module Yast2
+  class Group
+    class << self
```

Comment:
I'm not sure if I like to have so many class methods. In the past I used to use a similar approach, but now I feel is more correct to have a separate class for reading and querying, instead of a (ever growing) bunch of class methods.

Reply:
So having something like GroupReader? and GroupStorage? where the first has read and later has reset/all and then in yast2-users have also Writer?

Reply:
In network, we decided to separate reading/writing logic and, in the end, it looks like it was the right decision because it is easy to read/write the configuration from/to different sources. Of course, you can still keep class methods that hide those details.

Potentially, you could have a reader for `getent` and a separate one when `systemd-users` gets published :wink:.

Reply:
Yes, that was my point. The classes `User` and `Group` living in yast-yast2 should be basically structs, with almost zero logic. Nothing about reading/writing or even about connecting objects.

For example, this code in the class `User` is already a problem to me (it couples the shared class `User` with the particular reader that is based in getent (which may not be the only reader):

```ruby
def users
  # lazy load to avoid circular deps
  require "yast2/user"
  User.all.select { |u| u.gid == gid || users_name.include?(u.name) }
end
```